packages/block-library/src/tag-cloud/edit.js

```diff
@@ -54,6 +77,22 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 		return [ selectOption, ...taxonomyOptions ];
 	};
 
+	const onFontChange = ( fontSize, value ) => {
```

Comment:
We should use the current API for this parsing `__experimentalParseUnit` from components package. Also have in mind that with the merging of [this one](https://github.com/WordPress/gutenberg/pull/38987/), the function will be renamed.

Also we can't check with `indexOf` as it will show wrong results with things like `em and rem`. An implementation could be something like this:
```
const onFontSizeChange = ( fontSize, value ) => {
	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
	const [ quantity, newUnit ] = parseUnit( value );
	if ( ! Number.isFinite( quantity ) || quantity < 0 || quantity > 100 ) {
		return;
	}
	const updateObj = { [ fontSize ]: value };
	// We need to keep in sync the `unit` changes to both `smallestFontSize`
	// and `largestFontSize` attributes.
	Object.entries( {
		smallestFontSize,
		largestFontSize,
	} ).forEach( ( [ attribute, currentValue ] ) => {
		const [ _value, _unit ] = parseUnit( currentValue );
		// Only add an update if the other font size attribute has a different unit.
		if ( attribute !== fontSize && _unit !== newUnit ) {
			updateObj[ attribute ] = `${ _value }${ newUnit }`;
		}
	} );
	setAttributes( updateObj );
};
```

Reply:
whoa this code is much much better than mine. Thanks for this... learnt `Object.entries` 😋

Reply:
Wandering if instead of `quantity < 0 || quantity > 100` we could use the `min` and `max` props on `UnitControl` instead?

Reply:
Unfortunately this doesn't work well in all cases. This is the related issue: https://github.com/WordPress/gutenberg/issues/33291. We could still set some min/max values here though, but I guess we should add as `min` something like `0.1` maybe?

Reply:
Adding a `min` of `0.1` sounds sensible (just wondering if there's ever an instance where a user would like to set a value of `0`, but sounds like a hack that we shouldn't allow).

> This is the related issue: https://github.com/WordPress/gutenberg/issues/33291.

Thank you for the link! It's actually quite relevant in context of the recent TypeScript refactor that I'm attempting in https://github.com/WordPress/gutenberg/pull/38753